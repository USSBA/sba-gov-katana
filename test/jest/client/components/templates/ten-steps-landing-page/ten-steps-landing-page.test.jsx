import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import TenStepsLandingPage from 'templates/ten-steps-page/ten-steps-landing-page.jsx'

const store1 = {
  getState: () => {
    return false
  },
  subscribe: () => {
    return false
  },
  dispatch: () => {
    return false
  }
}
describe('TenStepsLandingPage', () => {
  test('should render TenStepsLandingPage', () => {
    Object.defineProperty(window.location, 'search', {
      writable: true,
      value: '?lang=en'
    })

    const component = shallow(<TenStepsLandingPage store={store1} />)
    expect(component).toMatchSnapshot()
  })
})
