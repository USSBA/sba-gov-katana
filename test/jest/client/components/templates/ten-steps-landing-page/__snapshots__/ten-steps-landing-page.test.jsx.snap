// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TenStepsLandingPage should render TenStepsLandingPage 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <TenStepsLandingPage
    actions={
        Object {
            "fetchContentIfNeeded": [Function],
          }
    }
    counsellorCta={undefined}
    menu={undefined}
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "subscribe": [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "notify": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": null,
          }
    }
/>,
  "nodes": Array [
    <TenStepsLandingPage
      actions={
            Object {
                  "fetchContentIfNeeded": [Function],
                }
      }
      counsellorCta={undefined}
      menu={undefined}
      store={
            Object {
                  "dispatch": [Function],
                  "getState": [Function],
                  "subscribe": [Function],
                }
      }
      storeSubscription={
            Subscription {
                  "listeners": Object {
                    "notify": [Function],
                  },
                  "onStateChange": [Function],
                  "parentSub": undefined,
                  "store": Object {
                    "dispatch": [Function],
                    "getState": [Function],
                    "subscribe": [Function],
                  },
                  "unsubscribe": null,
                }
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Connect(TenStepsLandingPage)
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "subscribe": [Function],
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Connect {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "store": undefined,
          "storeSubscription": undefined,
        },
        "notifyNestedSubs": [Function],
        "props": Object {
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "subscribe": [Function],
          },
        },
        "propsMode": true,
        "refs": Object {},
        "renderCount": 0,
        "selector": Object {
          "error": null,
          "props": Object {
            "actions": Object {
              "fetchContentIfNeeded": [Function],
            },
            "counsellorCta": undefined,
            "menu": undefined,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "subscribe": [Function],
            },
          },
          "run": [Function],
          "shouldComponentUpdate": false,
        },
        "setWrappedInstance": [Function],
        "state": Object {},
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "subscribe": [Function],
        },
        "subscription": Subscription {
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": null,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
        "version": 20,
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <TenStepsLandingPage
          actions={
                    Object {
                              "fetchContentIfNeeded": [Function],
                            }
          }
          counsellorCta={undefined}
          menu={undefined}
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "subscribe": [Function],
                            }
          }
          storeSubscription={
                    Subscription {
                              "listeners": Object {
                                "notify": [Function],
                              },
                              "onStateChange": [Function],
                              "parentSub": undefined,
                              "store": Object {
                                "dispatch": [Function],
                                "getState": [Function],
                                "subscribe": [Function],
                              },
                              "unsubscribe": null,
                            }
          }
/>,
        "_debugID": 2,
        "_renderedOutput": <TenStepsLandingPage
          actions={
                    Object {
                              "fetchContentIfNeeded": [Function],
                            }
          }
          counsellorCta={undefined}
          menu={undefined}
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "subscribe": [Function],
                            }
          }
          storeSubscription={
                    Subscription {
                              "listeners": Object {
                                "notify": [Function],
                              },
                              "onStateChange": [Function],
                              "parentSub": undefined,
                              "store": Object {
                                "dispatch": [Function],
                                "getState": [Function],
                                "subscribe": [Function],
                              },
                              "unsubscribe": null,
                            }
          }
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Connect(TenStepsLandingPage)
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "subscribe": [Function],
          }
    }
/>,
}
`;
